apps/src/templates/manageStudents/ManageStudentsTable.jsx

```diff
@@ -246,13 +246,16 @@ class ManageStudentsTable extends Component {
     let disabled = rowData.isEditing ?
       this.props.editingData[rowData.id].age.length === 0 :
       true;
+    const editedValue = rowData.isEditing ? this.props.editingData[rowData.id].sharingDisabled : true;
 
     return (
-      <span>
-        <Checkbox
-          disabled={disabled}
-        />
-      </span>
+      <ManageStudentsSharingCell
+        id={rowData.id}
+        isEditing={rowData.isEditing}
+        disabled={disabled}
+        checked={!rowData.sharingDisabled}
+        editedValue={editedValue}
```

Comment:
since your checked value is flipped before being passed to the component, I would also pass the edited value as flipped, just to make the logic within the component consistent/less confusing.